kuma/wiki/tests/test_managers.py

```diff
@@ -0,0 +1,251 @@
+from datetime import datetime
+import pytest
+
+from kuma.wiki.constants import REDIRECT_CONTENT
+from kuma.wiki.models import (Document, DocumentTag, LocalizationTag,
+                              ReviewTag, Revision)
+
+
+@pytest.fixture
+def redirect_doc(root_doc, wiki_user):
+    """A redirect document."""
+    html = REDIRECT_CONTENT % {'href': root_doc.get_absolute_url(),
+                               'title': root_doc.title}
+    doc = Document.objects.create(locale='en-US', slug='OldRoot', html=html)
+    doc.current_revision = Revision.objects.create(
+        document=doc,
+        creator=wiki_user,
+        content=html,
+        created=datetime(2017, 10, 27, 17, 4))
+    doc.save()
+    return doc
+
+
+@pytest.fixture
+def archive_doc(wiki_user):
+    """An archived document."""
+    archive = Document.objects.create(locale='en-US', slug='Archive',
+                                      title='Archive of obsolete content')
+    doc = Document.objects.create(locale='en-US', slug='Archive/Doc',
+                                  title='Archived document',
+                                  parent_topic=archive)
+    doc.current_revision = Revision.objects.create(
+        document=doc,
+        creator=wiki_user,
+        content='<p>Obsolete content.</p>',
+        comment='Obsolete content',
+        created=datetime(2017, 10, 27, 15, 48))
+    doc.save()
+    return doc
+
+
+def test_get_queryset(root_doc):
+    """Managers are customized for including / excluding deleted documents."""
+    deleted_doc = Document.objects.create(locale='en-US', slug='Deleted',
+                                          title='Deleted Document',
+                                          deleted=True)
+    assert Document.admin_objects.all().count() == 2
+    assert list(Document.objects.all()) == [root_doc]
+    assert list(Document.deleted_objects.all()) == [deleted_doc]
+
+
+def test_get_natural_key(root_doc):
+    """The locale + slug is the natural key for Documents."""
+    assert root_doc.natural_key() == (root_doc.locale, root_doc.slug)
+    assert root_doc == Document.objects.get_by_natural_key(root_doc.locale,
+                                                           root_doc.slug)
+
+
+@pytest.mark.parametrize(
+    'legacy_slug',
+    ('User:ethertank',
+     'Talk:Developer_Guide/Build_Instructions/Windows_Prerequisites',
+     'User_talk:ethertank',
+     'Template_talk:anch'))
```

Comment:
Nice use of `parametrize`, and seems nicer than creating a list of docs that includes all of the cases in one shot. I think a `Project_talk:` case needs to be added here as well?

Reply:
Oops missed that one. 👍